code/modules/assembly/holder.dm

```diff
@@ -62,54 +64,45 @@
 
 /obj/item/assembly_holder/update_overlays()
 	. = ..()
-	if(a_left)
-		. += "[a_left.icon_state]_left"
-		for(var/left_overlay in a_left.attached_overlays)
+	for(var/i in 1 to LAZYLEN(assemblies) step 2)
+		var/obj/item/assembly/assembly = assemblies[i]
+		. += "[assembly.icon_state]_left"
+		for(var/left_overlay in assembly.attached_overlays)
 			. += "[left_overlay]_l"
-
-	if(!a_right)
-		return
-
-	if(a_right.is_position_sensitive)
-		. += "[a_right.icon_state]_right"
-		for(var/right_overlay in a_right.attached_overlays)
-			. += "[right_overlay]_r"
-		return
-
-	var/mutable_appearance/right = mutable_appearance(icon, "[a_right.icon_state]_left")
-	right.transform = matrix(-1, 0, 0, 0, 1, 0)
-	for(var/right_overlay in a_right.attached_overlays)
-		right.add_overlay("[right_overlay]_l")
-	. += right
+	for(var/i in 2 to LAZYLEN(assemblies) step 2)
+		var/obj/item/assembly/assembly = assemblies[i]
+		var/mutable_appearance/right = mutable_appearance(icon, "[assembly.icon_state]_left")
+		right.transform = matrix(-1, 0, 0, 0, 1, 0)
+		for(var/right_overlay in assembly.attached_overlays)
+			right.add_overlay("[right_overlay]_l")
+		. += right
 
 /obj/item/assembly_holder/on_found(mob/finder)
-	if(a_left)
-		a_left.on_found(finder)
-	if(a_right)
-		a_right.on_found(finder)
+	for(var/obj/item/assembly/assembly in assemblies)
+		assembly.on_found(finder)
```

Comment:
`as anything`, if assemblies is guaranteed to only have /obj/item/assembly within.